### PR TITLE
Sanity-checking of some params

### PR DESCRIPTION
**Pull request by richvdh**
Some sanity-checking I added while I was trying to figure out where I was
getting an undefined event from.

### CODE REVIEW

File: lib/SyTest/Federation/Protocol.pm [hunk 1]
```diff
@@ -26,6 +26,7 @@ use Protocol::Matrix qw(
    encode_base64_unpadded
 );
 
+use Carp;
```
Comment (erikjohnston) [inline lib/SyTest/Federation/Protocol.pm]: What's Carp?

File: lib/SyTest/Federation/Room.pm [hunk 2]
```diff
@@ -385,14 +385,21 @@ sub make_join_protoevent
       $self->get_current_state_event( "m.room.create" ),
       $self->get_current_state_event( "m.room.join_rules" ),
    );
+   my $auth_events = $self->make_event_refs( @auth_events );
+
```
Comment (erikjohnston) [inline lib/SyTest/Federation/Room.pm]: Do we want to do the same checks for `auth_events`?

File: lib/SyTest/Federation/Protocol.pm (hunk 1 above)
Comment (richvdh) [reply lib/SyTest/Federation/Protocol.pm]: https://perldoc.perl.org/Carp.html, which exports `croak`.

File: lib/SyTest/Federation/Room.pm (hunk 2 above)
Comment (richvdh) [reply lib/SyTest/Federation/Room.pm]: I guess.

Comment (richvdh) [reply lib/SyTest/Federation/Room.pm]: actually no, that would be redundant. there's already a `grep defined` at line 384.

**Pull request merged**